assets/js/util/report-validation.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Reporting API validation utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Validates data that can be either string or object of the certain type, or array of them.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|string[]|Object|Object[]} data The data to check.
+ * @param {Function} objectVerificationCallback The callback to verify an object.
+ * @return {boolean} TRUE if data is valid, otherwise FALSE.
+ */
+function isValidStringsOrObjects( data, objectVerificationCallback ) {
```

Comment:
This isn't really a callback so much as a verification/check/test function. `verifyFunction` or `isValidObjectFunction` might work a bit better? But not a big deal.

Reply:
Updated.